.gitlab/includes/gcc12_cuda12_pipeline.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2023 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include:
+  - local: '.gitlab/includes/common_pipeline.yml'
+  - local: '.gitlab/includes/common_spack_pipeline.yml'
+
+.variables_gcc12_cuda12_config:
+  variables:
+    ARCH: linux-ubuntu22.04-zen3
+    BUILD_TYPE: Debug
+    COMPILER: gcc@12.1.0
+    CXXSTD: 17
```

Comment:
Perhaps add a `GPU_TARGET` like you've done for HIP and reuse it for the CMake flags later. I don't think it affects our dependencies at the moment, but better safe than sorry.